pyuvdata/uvcal/uvcal.py

```diff
@@ -633,22 +737,22 @@ def new(**kwargs):
         """
         return initializers.new_uvcal(**kwargs)
 
-    def _set_flex_spw(self):
+    def _set_flex_spw(self, flex_spw=True):
         """
         Set flex_spw to True, and adjust required parameters.
 
         This method should not be called directly by users; instead it is called
         by the file-reading methods to indicate that an object has multiple spectral
         windows concatenated together across the frequency axis.
         """
-        assert (
-            not self.wide_band
+        assert not (
+            self.wide_band and flex_spw
         ), "Cannot set objects wide_band objects to have flexible spectral windows."
         # Mark once-optional arrays as now required
-        self.flex_spw = True
-        self._flex_spw_id_array.required = True
+        self.flex_spw = flex_spw
+        self._flex_spw_id_array.required = flex_spw
         # Now make sure that chan_width is set to be an array
-        self._channel_width.form = ("Nfreqs",)
+        self._channel_width.form = ("Nfreqs",) if flex_spw else ()
```

Comment:
Why add the option to make it not flex spw? In Version 3.0 we're always going to require the `flex_spw_id_array` (and the flex_spw attribute will either go away or be deprecated).

Reply:
Removed in 766e32646768181b70ddd8d2e576d90ca57c254c.